### PR TITLE
State process test

### PR DESCRIPTION
**Pull request by boundless-forest**
Add test cases to ensure the state process correctness. Those tests only apply to the Crab and Crab Parachain states now.

### CODE REVIEW

File: tool/state-processor/src/system/mod.rs [hunk 1]
```diff
@@ -211,8 +211,12 @@ impl Processor {
 		account_infos.iter_mut().for_each(|(_, v)| v.data.adjust());
 
 		log::info!("merge solo remaining balances");
-		remaining_ring.into_iter().for_each(|(k, v)| {
+		let (mut total_remaining_ring, mut total_remaining_kton) =
+			(u128::default(), u128::default());
+
+		remaining_ring.into_iter().filter(|(_, v)| *v > 0).for_each(|(k, v)| {
```
Comment (AurevoirXavier) [inline tool/state-processor/src/system/mod.rs:217]: I think the filter is unnecessary.

Comment (boundless-forest) [reply tool/state-processor/src/system/mod.rs:217]: Why? Most of the items' account under this prefix is zero. We don't need to handle them in mineview.

Comment (AurevoirXavier) [reply tool/state-processor/src/system/mod.rs:217]: It doesn't make any changes to the following logic.

File: tool/state-processor/src/indices/mod.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ impl Processor {
 
 		// https://github.dev/darwinia-network/substrate/blob/darwinia-v0.12.5/frame/indices/src/lib.rs#L154
 		log::info!("adjust the reserved's decimals then free solo `Indices` reservations");
-		accounts.into_iter().for_each(|(_, (a, mut v, _))| {
+		accounts.into_iter().filter(|(_, (_, v, _))| *v > 0).for_each(|(_, (a, mut v, _))| {
```
Comment (AurevoirXavier) [inline tool/state-processor/src/indices/mod.rs:15]: here

File: tool/state-processor/src/system/mod.rs [hunk 3]
```diff
@@ -221,16 +225,19 @@ impl Processor {
 				);
 			}
 		});
-		remaining_kton.into_iter().for_each(|(k, v)| {
+		remaining_kton.into_iter().filter(|(_, v)| *v > 0).for_each(|(k, v)| {
```
Comment (AurevoirXavier) [inline tool/state-processor/src/system/mod.rs:228]: here

**Pull request merged**